odoo/addons/base/models/ir_attachment.py

```diff
@@ -495,7 +495,9 @@ def write(self, vals):
 
     def copy(self, default=None):
         self.check('write')
-        default = dict(default or {}, datas=self.datas)
+        if not set(default or {}) & set(['datas', 'db_datas']):
```

Comment:
What about adding 'raw' in the second set? otherwise we can't duplicate attachment while overwriting `raw`.
(useful to avoid the base64encode & decode)

Reply:
It's planned in the master version but the field did not exist in 13.0

Reply:
The RHS can use a set literal though. And the LHS could be `(default or {}).keys()` to avoid copying to a new set.